components/Map.js

```diff
@@ -59,93 +116,159 @@ export default function Map({selected, selectedDest, radius}) {
         }
       );
     }
-  }, []);
-
-  const onLoadMap = map => {
-    setMap(map);
   };
 
-  const onLoadStartMarker = () => {
-    const request = {
-      location: selected,
-      radius: radius,
-      type: ["bar"],
-    };
+  const getNearbyBars = (location, map) => {
+    if (!location || !radius) return;
     const service = new google.maps.places.PlacesService(map);
-    service.nearbySearch(request, callback);
-
-    function callback(results, status) {
-      if (status == google.maps.places.PlacesServiceStatus.OK) {
-        console.log(results);
-        setBars(
-          results.filter(
-            bar =>
-              !bar.types.includes("restaurant") &&
-              bar.business_status === "OPERATIONAL"
-          )
+    service.nearbySearch(
+      {location: location, radius: radius, type: ["bar"]},
+      (results, status, pagination) => {
+        if (status !== "OK" || !results) return;
+        const filteredBars = results.filter(
+          bar =>
+            !bar.types.includes("restaurant") &&
+            bar.business_status === "OPERATIONAL"
         );
+        setSearchresults(filteredBars);
+        setPagination(pagination);
       }
+    );
+  };
+
+  const getWaypoints = () => {
+    if (bars.length === 0) return;
+
+    const getWaypointsFromBars = bar => {
+      const lat = bar.geometry.location.lat();
+      const lng = bar.geometry.location.lng();
+      const location = {lat: lat, lng: lng};
+      if (
+        !waypoints.some(
+          waypoint =>
+            waypoint.location.lat === lat && waypoint.location.lng === lng
+        )
+      ) {
+        setWaypoints(waypoints => [
+          ...waypoints,
+          {location: location, stopover: true},
+        ]);
+      }
+    };
+    if (stops < bars.length) {
+      bars.slice(0, stops).forEach(bar => getWaypointsFromBars(bar));
+    } else if (stops > bars.length) {
+      alert(
+        "Not enough bars found in your area. Please increase radius or lower the amount of stops!"
+      );
+    } else {
+      bars.forEach(bar => getWaypointsFromBars(bar));
     }
   };
 
-  const handleMarkerClicked = id => {
-    setMarkerClicked(id);
+  const calculateRoute = async () => {
+    if (waypoints.length === 0) return;
+    const directionsService = new google.maps.DirectionsService();
+    const results = await directionsService.route({
+      origin: selected,
+      destination: selectedDest,
+      waypoints: waypoints,
+      optimizeWaypoints: true,
+      travelMode: google.maps.TravelMode.WALKING,
+    });
+    setDirectionsResponse(results);
   };
 
+  const logfunction = () => {
+    console.log("bars");
+    console.log(bars);
+    console.log("directionsResponse");
+    console.log(directionsResponse);
+    console.log("waypoints");
+    console.log(waypoints);
+    console.log(directionsResponse.routes[0].legs[0].distance.text);
```

Comment:
You really need this?

Reply:
Woops. No, I just had this to check my various states. 
It was super useful for debugging and refactoring. I will remove it